src/main/java/com/google/sps/servlets/AddSaleCardServlet.java

```diff
@@ -99,44 +107,75 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-      // TODO: Create vendor key using the Auth ID
-      // Key vendorKey = KeyFactory.createKey("Vendor", (String) user.authID);
-
-      // Generate Key when datastore put
-      Entity saleCard = new Entity("SaleCard"/*, vendorKey*/);
-      datastore.put(saleCard);      
-
-      // TODO: Remove from datastore existing saleCard, picture and location data
+      Key vendorKey = KeyFactory.createKey("Vendor", userService.getCurrentUser().getUserId());
+      Entity vendorEntity;
+      try {
+        vendorEntity = datastore.get(vendorKey);
+      } catch (EntityNotFoundException e) {
+        System.out.println(e);
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      Vendor vendorObject = new Vendor(vendorEntity);
+
+      Entity saleCard;
+      Entity picture;
+      Entity locationData;
+      if (vendorObject.getSaleCard() == null) {
+        // New Entities - (Generate Key when datastore put)
```

Comment:
I think the comment should be before the if statement

Reply:
Done!